src/main/java/net/pms/dlna/DLNAMediaAudio.java

```diff
@@ -712,19 +728,62 @@ public void setSampleFrequency(String sampleFrequency) {
 	 *
 	 * @return The number of channels
 	 * @since 1.50
-	 * @deprecated Use getAudioProperties().getNumberOfChannels() instead
+	 * @deprecated Use {@link #getNumberOfChannels()} instead.
 	 */
 	@Deprecated
 	public int getNrAudioChannels() {
-		return audioProperties.getNumberOfChannels();
+		return getNumberOfChannels();
+	}
+
+	/**
+	 * Returns an array with the number of audio channels, with the "base" layer
+	 * at index 0.
+	 *
+	 * @return The array of audio channels or an array containing
+	 *         {@link #NUMBEROFCHANNELS_DEFAULT} if unknown.
+	 */
+	public int[] getNumberOfChannelsArray() {
+		if (
+			numberOfChannelsArray != null &&
+			numberOfChannelsArray.length > 0 &&
+			UMSUtils.getIntArrayMaxValue(numberOfChannelsArray, 0) > 0
+		) {
+			return numberOfChannelsArray;
+		}
+		return new int[] {getNumberOfChannels()};
+	}
+
+	/**
+	 * Returns the highest number of audio channels if there are multiple
+	 * values, otherwise the single value.
+	 *
+	 * @return The number of channels or {@link #NUMBEROFCHANNELS_DEFAULT} if
+	 *         unknown.
+	 * @since 6.7.2
+	 */
+	@SuppressWarnings("deprecation")
+	public int getNumberOfChannels() {
+		if (numberOfChannelsArray != null && numberOfChannelsArray.length > 0) {
+			int highest = UMSUtils.getIntArrayMaxValue(numberOfChannelsArray, 0);
+			if (highest > 0) {
+				return highest;
+			}
+		}
+		if (nrAudioChannels > 0) {
+			return nrAudioChannels;
+		}
+		if (audioProperties != null && audioProperties.getNumberOfChannels() > 0) {
```

Comment:
@Sami32 I don't know, checking the known plugins takes some effort which I'm not keen on doing for every change I make. There might be plugins we're not aware of as well. I'm simply trying to make as little problems as possible, even though I realize that much of this probably isn't used by any code. You can bet though, that the one time I forget or it's not viable to make something backwards compatible, something will break 😉